### PR TITLE
[huggingface_tensorflow] upgrade Tensorflow to 2.6 

### PR DESCRIPTION
**Pull request by philschmid**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

Splits into separate PRs 
#1582 

@arjkesh 

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [x] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [x] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [x] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [x] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [x] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: huggingface/tensorflow/inference/docker/2.6/py3/cu112/Dockerfile.gpu [hunk 1]
```diff
@@ -0,0 +1,130 @@
+FROM nvidia/cuda:11.3.1-base-ubuntu20.04
```
Comment (arjkesh) [inline huggingface/tensorflow/inference/docker/2.6/py3/cu112/Dockerfile.gpu:1]: @philschmid - these are marked as cuda 11.3, but the directory and tagging structure is 11.2. Is there a reason why you used 11.3 here? I'd suggest using 11.2 if possible, since that is in line with the training images

**Comment by Qingzi-Lan** [comment]
from sanity test, we need
`transformers` upgrade from 4.16.2 to 4.17.0
`tensorflow` upgrade from 2.6.2 to 2.6.3
`numpy` upgrade from 1.19.5 to 1.22.2
will these upgrade affect any functionality? For TF version, I believe it's pined to 2.6.2.

**Comment by Qingzi-Lan** [comment]
let's wait TF2.6.3 base images update, and then retry the build

**Comment by Qingzi-Lan** [comment]
@philschmid why do we need typing-extensions version pin to 3.7.4.3 here

**Comment by philschmid** [comment]
> @philschmid why do we need typing-extensions version pin to 3.7.4.3 here

tensorflow 2.6.0 depends on typing-extensions~=3.7.4 and the `huggingface_hub` on `typing-extensions>=3.7.4.3` [[REF](https://github.com/huggingface/huggingface_hub/blob/a002a53440067d6f748e13d7c62df55d149b284f/setup.py#L19)]. 
I tried to use `~=3.7.4` but it seems to generate an error that's why I pinned it.

File: huggingface/tensorflow/training/docker/2.6/py3/cu112/Dockerfile.gpu [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+RUN pip install --no-cache-dir \
+    transformers[sklearn,sentencepiece]==${TRANSFORMERS_VERSION} \
+    datasets==${DATASETS_VERSION} \
+    tensorflow-addons==0.12.0 \
+    psutil \
+    # TODO: Remove upgrade statements once packages are updated in base image
+    "Pillow>=9.0.0" \
+    "typing-extensions==3.7.4.3" \
+    "bokeh==2.3.3"
+
```
Comment (arjkesh) [inline huggingface/tensorflow/training/docker/2.6/py3/cu112/Dockerfile.gpu:29]: I imagine most of these are fixed in TF 2.6.3 since it was just released, can we try removing some of these temporary packages?

Comment (Qingzi-Lan) [reply huggingface/tensorflow/training/docker/2.6/py3/cu112/Dockerfile.gpu:29]: removed Pillow 9.0.1 since it's already installed in the base image
the other two are for HF specifically, modified the comment for it.

**Review by saimidu — COMMENTED** [review]
LGTM. Please run the `rc` tests as well, as a final validation step, before merging this PR.

**Review by saimidu — APPROVED** [review]
